osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -221,8 +221,14 @@ private void load(OsuConfigManager config, OsuColour colours, ChannelManager cha
                 // TODO: consider scheduling bindable callbacks to not perform when overlay is not present.
                 channelManager.JoinedChannels.ItemsAdded += onChannelAddedToJoinedChannels;
                 channelManager.JoinedChannels.ItemsRemoved += onChannelRemovedFromJoinedChannels;
+
+                bool channelSelected = channelManager.CurrentChannel.Value != null;
+
                 foreach (Channel channel in channelManager.JoinedChannels)
-                    ChannelTabControl.AddChannel(channel);
+                    ChannelTabControl.AddChannel(channel, !channelSelected);
+
+                if (channelSelected)
+                    ChannelTabControl.Current.Value = channelManager.CurrentChannel.Value;
```

Comment:
This logic is quite convoluted. You're handling the "no channel selected" case in `ChannelTabControl` and the "channel selected" case here, which is going to be confusing to whoever is going to be looking for the place where the channel can change down the line. Should handle both here and make `AddChannel` not change tabs if anything, IMO.

Reply:
Are these changes fine?
```diff
@ -222,13 +222,10 @@ private void load(OsuConfigManager config, OsuColour colours, ChannelManager cha
                channelManager.JoinedChannels.ItemsAdded += onChannelAddedToJoinedChannels;
                channelManager.JoinedChannels.ItemsRemoved += onChannelRemovedFromJoinedChannels;

-               bool channelSelected = channelManager.CurrentChannel.Value != null;

                foreach (Channel channel in channelManager.JoinedChannels)
-                   ChannelTabControl.AddChannel(channel, !channelSelected);
+                   ChannelTabControl.AddChannel(channel);

-               if (channelSelected)
-                   ChannelTabControl.Current.Value = channelManager.CurrentChannel.Value;
+               ChannelTabControl.Current.Value = channelManager.CurrentChannel.Value == null ? channelManager.JoinedChannels.First() : channelManager.CurrentChannel.Value;

                channelManager.AvailableChannels.ItemsAdded += availableChannelsChanged;
                channelManager.AvailableChannels.ItemsRemoved += availableChannelsChanged;
```

```diff
@ -59,17 +59,12 @@ protected override TabItem<Channel> CreateTabItem(Channel value)

        /// <summary>
        /// Adds a channel to the ChannelTabControl.
-       /// The first channel added will automaticly selected if <paramref name="setChannel"/> is true.
        /// </summary>
        /// <param name="channel">The channel that is going to be added.</param>
-       /// <param name="setChannel">If the current channel should be changed if none was selected before</param>
-       public void AddChannel(Channel channel, bool setChannel = true)
+       public void AddChannel(Channel channel)
        {
            if (!Items.Contains(channel))
                AddItem(channel);
-
-           if (Current.Value == null && setChannel)
-               Current.Value = channel;
        }

        /// <summary>
```

Reply:
That is what I had in mind, so it looks good to me, yes.